### PR TITLE
Ch 773

### PR DESCRIPTION
**Pull request by eshtadc**

### CODE REVIEW

File: personalize.admin.inc [hunk 1]
```diff
@@ -646,6 +646,11 @@ function personalize_agent_page($agent_data) {
     ),
   );
   $agent_instance = personalize_agent_load_agent($agent_data->machine_name);
+  if ($agent_instance instanceof PersonalizePageVariationInterface) {
+    $build['return_link'] = array(
+      '#markup' => l(t('Go to %campaign campaign', array('%campaign' => $agent_data->label)), personalize_page_variation_get_path($agent_data->machine_name, $agent_data->machine_name), array('html' => TRUE)),
```
Comment (katbailey) [inline personalize.admin.inc]: The assumption that the variation set name is the same as the agent name might be problematic here - that happens to be what the Acquia Lift Simple A/B agent does, but you could in theory have a variation set that is not named the same as the agent, right?

File: personalize.admin.inc [hunk 2]
```diff
@@ -709,6 +717,10 @@ function personalize_agent_form($form, &$form_state, $agent_data = NULL) {
       }
     }
   }
+
+  // Allow the campaign end date to be set.
+  $form['agent_form']['agent_fieldset']['campaign_end'] = personalize_agent_end_date_form(!empty($agent_data->machine_name) ? $agent_data->machine_name : NULL);
+
```
Comment (katbailey) [inline personalize.admin.inc:713]: This change seems unrelated - does this belong here?

File: personalize.admin.inc (hunk 1 above)
Comment (katbailey) [inline personalize.admin.inc]: Maybe we could invoke an alter hook here passing in the build array by reference, and acquia_lift module could implement that hook to add this link? It could also just form_alter the various forms to make whatever changes it needs to. It just feels like we're adding complexity to the base module for the sake of one crazy-ass use case that we need to support in Acquia Lift module :/ Though I do realize that I started us down that path by adding the personalize_create_page_variation() function here in the first place. I dunno, what do you think?

File: personalize.admin.inc (hunk 2 above)
Comment (eshtadc) [inline personalize.admin.inc:713]: I was adding in the functionality while working on changes to the form.... (came from a UX review).  I could certainly pull it into its own commit and tack it on and create an issue to expose the end date via the UI (or tack it on to the issue where you added the functionality in the first place).

File: personalize.admin.inc [hunk 3]
```diff
@@ -646,6 +646,11 @@ function personalize_agent_page($agent_data) {
     ),
   );
   $agent_instance = personalize_agent_load_agent($agent_data->machine_name);
+  if ($agent_instance instanceof PersonalizePageVariationInterface) {
```
Comment (eshtadc) [inline personalize.admin.inc]: Also - need to remove acquia_lift_agent_edit_callback since it is expected that they can come to this page now....

File: personalize.admin.inc (hunk 1 above)
Comment (eshtadc) [inline personalize.admin.inc]: I like the form alter idea.  The weirdness will be that acquia_lift module will turn around and call personalize to get the variations link... but at least its somewhat less twisted than it is now.  Agreed about the variation_set_name/agent_name assumption since we have the interface we're checking against instead of just the specific type (forgot I changed that).... Originally I had the interface include a method to get the pageVariationPath - guess I can add that back in and let the simpleAB class implement it (with the personalize_ function).  Still seems like weird gymnastics going back and forth across the modules like that.... can you think of anything cleaner?

File: personalize.admin.inc (hunk 2 above)
Comment (katbailey) [inline personalize.admin.inc:713]: Ah no, let's leave it here - I just thought it might have shown up here from a dodgy rebase or something :P

File: personalize.admin.inc (hunk 1 above)
Comment (katbailey) [inline personalize.admin.inc]: "The weirdness will be that acquia_lift module will turn around and call personalize to get the variations link"
Do you mean the personalize_page_variation_get_path() function? Maybe we don't need that at all, we just move all that logic back over to acquia_lift..?

Comment (eshtadc) [inline personalize.admin.inc]: You mean all of the personalize_page_variation_.... functions?  I could see that - personalize would still have the interface but would leave details of how to implement a pagevariation campaign to the module implementing it... makes sense.
